test/integration/test_path.py

```diff
@@ -101,3 +103,52 @@ def test_write_delete(backend_aware_bucketfs_params, children_poem, classic_poem
     poem_path1.rm()
     expected_names = {"classic", "highlands.txt"}
     assert _collect_all_names(poems_root) == expected_names
+
+
+def test_infer_path_onprem(backend_aware_bucketfs_params):
```

Comment:
backend_aware_bucketfs_params returns SaaS or on-prem parameters, so if you specifically want to test on-prem, you need to skip SaaS. Alternatively, we could also think about, only having one test and check on the inferred path, if we can upload files to the path.

Reply:
I vote to keep two tests, as the checks in one test would become more complex.....